frontends/ol-search-ui/src/util.tsx

```diff
@@ -144,3 +155,76 @@ export const CertificateIcon = () => (
     src="/static/images/certificate_icon_infinite.png"
   />
 )
+
+export const minPrice = (
+  prices: Array<CoursePrice>,
+  includeDollarSign = false
+) => {
+  if (emptyOrNil(prices)) {
+    return null
+  }
+  const price = Math.min(...prices.map(price => price.price))
+
+  if (price > 0 && price !== Infinity) {
+    return includeDollarSign ? `${formatPrice(price)}` : price
+  } else {
+    return "Free"
+  }
+}
+
+export const getStartDate = (
+  object: LearningResourceResult,
+  objectRun: LearningResourceRun
+): string => {
+  if (object.platform === "ocw") {
+    return `${capitalize(objectRun.semester || "")} ${objectRun.year || ""}`
+  } else if (objectRun.start_date) {
+    return moment(objectRun.start_date).format("MMMM DD, YYYY")
+  } else if (objectRun.best_start_date) {
+    return moment(objectRun.best_start_date).format("MMMM DD, YYYY")
+  }
+  return "Ongoing"
+}
+
+export const getInstructorName = (instructor: CourseInstructor) => {
+  if (instructor.full_name) {
+    return instructor.full_name
+  } else if (instructor.first_name && instructor.last_name) {
+    return `Prof. ${instructor.first_name} ${instructor.last_name}`
+  } else if (instructor.last_name) {
+    return `Prof. ${instructor.last_name}`
+  }
+  return ""
+}
+
+export const emptyOrNil = R.either(R.isEmpty, R.isNil)
+
+export const languageName = (langCode: string | null): string =>
+  LocaleCode.getLanguageName(
+    `${langCode ? langCode.split("-")[0].toLowerCase() : "en"}-US`
+  )
+
+const formatPrice = (price: number | null | undefined): string => {
+  if (price === null || price === undefined) {
+    return ""
+  } else {
+    const decimalPrice = new Decimal(price)
+    let formattedPrice
+
+    if (decimalPrice.isInteger()) {
+      formattedPrice = decimalPrice.toFixed(0)
+    } else {
+      formattedPrice = decimalPrice.toFixed(2, Decimal.ROUND_HALF_UP)
+    }
+    return `$${formattedPrice}`
+  }
+}
+
+export const absolutizeURL = (url: string) =>
+  new URL(url, window.location.origin).toString()
+
+// @ts-expect-error typescript complains about getting 0 arguments
+export const capitalize = R.converge(R.concat(), [
+  R.compose(R.toUpper, R.head),
+  R.tail
+]) as F.Curry<(text: string) => string>
```

Comment:
Does this just capitalize the first letter? And if it does, would
```ts
const capitalize = (s: string): string => s[0].toUpperCase() + s.slice(1)
```
work ?

Reply:
The above breaks for empty strings